File: swift/sprint5/G/solution.swift
#if !REMOTE_JUDGE
open class Node {

    public var value: Int
    public var left: Node?
    public var right: Node?

    public init(_ value: Int) {
        self.value = value
        self.right = nil
        self.left = nil
    }
}
#endif

func solution(node: Node?) -> Int {
    // Your code
    // ヽ(´▽`)/
    return 0
}

func test() {
    var node1 = Node(5)
    var node2 = Node(1)
    var node3 = Node(-3)
    node3.left = node2
    node3.right = node1
    var node4 = Node(2)
    var node5 = Node(2)
    node5.left = node4
    node5.right = node3
    assert(solution(node: node5) == 6)
}

File: csharp/sprint5/G/Solution.cs
#if !REMOTE_JUDGE
public class Node
{
    public int Value { get; set; }
    public Node Left { get; set; }
    public Node Right { get; set; }

    public Node(int value)
    {
        Value = value;
        Left = null;
        Right = null;
    }
}
#endif

public class Solution
{
    public static int Solve(Node root)
    {
        // Your code
        // “ヽ(´▽｀)ノ”
    }

    #if !REMOTE_JUDGE
    private static void Main()
    {
        var node1 = new Node(5);
        var node2 = new Node(1);
        var node3 = new Node(-3)
        {
            Left = node2,
            Right = node1,
        };

        var node4 = new Node(2);
        var node5 = new Node(2)
        {
            Left = node4,
            Right = node3
        };

        System.Console.WriteLine(Solve(node5) == 6);
    }
    #endif
}
